clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/EventHandler.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.consumer.internals.events;
+
+/**
+ * This class interfaces with the KafkaConsumer and the background thread. It allows the caller to enqueue events via
+ * the {@code add()} method and to retrieve events via the {@code poll()} method.
+ */
+public interface EventHandler {
+    public BackgroundEvent poll();
```

Comment:
Can we add some javadoc for this? Does this return null if there are no events available?

Reply:
Thanks @hachikuji - I'm making it to return an Optional.  Another question raised here, I think we need to handle the exceptions thrown by blockingQueue.add.  I'm looking at the RaftMessageQueue impl, the exceptions aren't being handled there, are we expecting the caller/client to handle these exceptions?